lib/memery.rb

```diff
@@ -53,6 +56,7 @@ def memoized?(method_name)
     def prepend_memery_module!
       return if defined?(@_memery_module)
       @_memery_module = Module.new { extend MemoizationModule }
+      @_memery_module.use_hashed_arguments = memery_use_hashed_arguments
```

Comment:
I guess this way setting `self.memery_use_hashed_arguments = true` will only work in case when you call it before making any `memoize` call. This might be not very clear, maybe it would be better to either

1. Store it on the top level (`Memery` module).
2. Get it directly from the class and not from the `@_memery_module`? So that we always get the up-to-date value.